CMakeLists.txt

```diff
@@ -152,7 +152,7 @@ set(COMPILER_SUPPORT_LIBRARIES "")
 
 # If we are using MSVC, we need /bigobj.
 if (MSVC)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj /Zm400")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj /Zm400 /Zc:__cplusplus")
```

Comment:
I think there is one other piece: we need to make sure that `/std:c++17` is set.  Maybe that is done by another part of the CMake configuration?  The default `__cplusplus` value is `201402L` if no `/std:` flag is set on MSVC.

Reply:
I thought about this, I use the standard `set(CMAKE _CXX_STANDARD)` for CMake let us see if this passes, otherwise I will re-add the `/std:c++17` switch

Reply:
Agreed, but maybe for whatever reason that is not forcing C++17 in MSVC?  You could check the build logs and see what the exact compilation flags being used are, I suppose.